### PR TITLE
Add exclusions feature

### PR DESCRIPTION
**Pull request by exoszajzbuk**
- exclusions can be set
- excluded files/directories will be skipped during upload

### CODE REVIEW

File: lib/capistrano/s3/publisher.rb [hunk 1]
```diff
@@ -78,8 +78,8 @@ def self.base_file_path(root, file)
           file.gsub(root, "")
         end
 
-        def self.files(deployment_path)
-          Dir.glob("#{deployment_path}/**/*")
+        def self.files(deployment_path, exclusions)
+          Dir.glob("#{deployment_path}/**/*") - Dir.glob(exclusions.map{|e| "#{deployment_path}/#{e}"})
```
Comment (j15e) [inline lib/capistrano/s3/publisher.rb]: I would use spaces around } { for readability

`exclusions.map { |e| "#{deployment_path}/#{e}" }`

https://github.com/bbatsov/ruby-style-guide#spaces-braces

**Comment by exoszajzbuk** [comment]
I'm not sure why it is suddenly failing after a tiny code-style fix :)

**Comment by j15e** [comment]
After rubinius, jruby :trollface: 

I'll see what I can do

**Comment by j15e** [comment]
I will merge & fix build in master, it is indeed unrelated to your changes.

**Comment by j15e** [comment]
There was an issue with jruby & bundler 1.14.2 which was fixed in 1.14.3 released just a few minutes ago. Master build fixed.

https://github.com/bundler/bundler/blob/master/CHANGELOG.md

**Pull request merged**